lib/cisco_node_utils/radius_server.rb

```diff
@@ -0,0 +1,294 @@
+# Radius Server provider class
+
+# Jonathan Tripathy et al., September 2015
+
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # RadiusServer - node utility class for
+  # Raidus Server configuration management
+  class RadiusServer < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      unless name[/^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/]
+        fail ArgumentError, 'Invalid value (Name is not an IP address)'
+      end
+      @name = name
+
+      create if instantiate
+    end
+
+    def self.radiusservers
+      hash = {}
+
+      radiusservers_list = config_get('radius_server', 'hosts')
+      return hash if radiusservers_list.nil?
+
+      radiusservers_list.each do |id|
+        hash[id] = RadiusServer.new(id, false)
+      end
+
+      hash
+    end
+
+    def create
+      config_set('radius_server',
+                 'hosts',
+                 state: '',
+                 ip:    @name)
+    end
+
+    def destroy
+      config_set('radius_server',
+                 'hosts',
+                 state: 'no',
+                 ip:    @name)
+    end
+
+    def ==(other)
+      name == other.name
+    end
+
+    def auth_port
+      val = config_get('radius_server', 'auth-port', @name)
+      val = val[0].to_i unless val.nil?
+      val
+    end
+
+    def auth_port_default
```

Comment:
In order to be able to use the autogen logic more effectively, all of the default-getter methods should be named `default_foo` rather than `foo_default`.